app/routers/dataresource.py

```diff
@@ -117,120 +97,56 @@ async def read_dataresource(
 
     Parse data information using the appropriate parser.
     """
-    if not await cache.exists(resource_id):
-        raise httpexception_404_item_id_does_not_exist(resource_id, "resource_id")
-    if session_id and not await cache.exists(session_id):
-        raise httpexception_404_item_id_does_not_exist(session_id, "session_id")
-
-    cache_value = await cache.get(resource_id)
-    if not isinstance(cache_value, (str, bytes)):
-        raise TypeError(
-            f"Expected cache value of {resource_id} to be a string or bytes, "
-            f"found it to be of type {type(cache_value)!r}."
-        )
+    cache_value = await _fetch_cache_value(cache, resource_id, "resource_id")
     config = ResourceConfig(**json.loads(cache_value))
 
     if session_id:
-        cache_value = await cache.get(session_id)
-        if not isinstance(cache_value, (str, bytes)):
-            raise TypeError(
-                f"Expected cache value of {session_id} to be a string or bytes, "
-                f"found it to be of type {type(cache_value)!r}."
-            )
-    session_data: "Optional[dict[str, Any]]" = (
-        None if not session_id else json.loads(cache_value)
-    )
-
-    if config.downloadUrl and config.mediaType:
-        # Download strategy
-        session_update = create_strategy("download", config).get(session=session_data)
-        if session_update and session_id:
-            session_data = await _update_session(session_id, session_update, cache)
+        session_data = await _fetch_cache_value(cache, session_id, "session_id")
+        populate_config_from_session(json.loads(session_data), config)
 
-        # Parse strategy
-        session_update = create_strategy("parse", config).get(session=session_data)
-        if session_update and session_id:
-            await _update_session(session_id, session_update, cache)
+    if not config.resourceType:
+        raise httpexception_422_resource_id_is_unprocessable(resource_id)
 
-    elif config.accessUrl and config.accessService:
-        # Resource strategy
-        session_update = create_strategy("resource", config).get(session=session_data)
+    if (config.downloadUrl and config.mediaType) or (
+        config.accessUrl and config.accessService
+    ):
+        session_update = create_strategy("resource", config).get()
         if session_update and session_id:
-            await _update_session(session_id, session_update, cache)
-
+            await _update_session(
+                session_id=session_id, updated_session=session_update, redis=cache
+            )
     else:
         raise httpexception_422_resource_id_is_unprocessable(resource_id)
```

Comment:
Remove this and instead add the negative parts of the `if` check alongside the check for the existance of `resourceType` above.

Reply:
Actually, then there is no need to do the current `if` check, and the `session_update = create_strategy...` lines can be de-indented.